fhir-persistence-schema/docs/SchemaDesign.md

```diff
@@ -0,0 +1,650 @@
+| parameter_names | Normalized data | Search parameter names. |
+| common_token_values | Normalized data | Normalized token and code system values. |
+| common_canonical_values | Normalized data | Normalized canonical values. |
+| logical_resources | Whole system | One row for each logical resource, regardless of type. |
+| resource_change_log | Whole system | One row for each CREATE, UPDATE or DELETE interaction. |
+| xx_logical_resources | Resource | Resource-type specific entry for each logical resource. |
+| xx_resources | Resource version | Resource-type specific entry for each version of a logical resource, including the resource payload unless payload offloading has been configured. |
+| xx_str_values     | Resource parameters | String search parameters, with indexes for both plain and lower case (normalized) variants. |
+| xx_number_values   | Resource parameters | Number search parameter values. |
+| xx_quantity_values | Resource parameters | Quantity search parameter values. |
+| xx_date_values     | Resource parameters | Date search parameter values. |
+| xx_latlng_values   | Resource parameters | Latitude/longitude search parameter values. |
+| xx_resource_token_refs       | Resource parameters | Token search parameter values. The token values are normalized to save space from repeating long strings. This table acts as a mapping table between the normalized values stored in COMMON_TOKEN_VALUES and the logical resource. |
+| xx_tags           | Resource parameters | Tag search parameters. Tag is a search parameter (rather than a type of search parameter) so no reference to PARAMETER_NAMES is needed. Tags are token values which are normalized in the COMMON_TOKEN_VALUES table. This table acts as a mapping between the normalized value and the logical resource. |
+| xx_profiles           | Resource parameters | Profile search parameters. Profile is a search parameter (rather than a type of search parameter) so no reference to PARAMETER_NAMES is needed. |
+| xx_security           | Resource parameters | Security search parameters. Security is a search parameter (rather than a type of search parameter) so no reference to PARAMETER_NAMES is needed. |
+| xx_current_refs       | Resource parameters | Problems, Medications, Allergies, Drug allergies current lists. |
+| str_values                 | Whole system parameters | Whole-system search string values. |
+| date_values                | Whole system parameters | Whole-system search date values. |
+| resource_token_refs        | Whole system parameters | Whole-system search token values. Mapping table to common_token_values. |
+| logical_resource_profiles  | Whole system parameters | Whole-system search profile values. |
+| logical_resource_tags      | Whole system parameters | Whole-system search tag values.                |
+| logical_resource_security  | Whole system parameters | Whole-system search security values            |
```

Comment:
Should the Erase Table be mentioned here? There was a new long running table for persistence offloading. I couldn't find it when I did my local search.

Reply:
added `erased_resources`